lib/shoulda/matchers/active_model/validate_presence_of_matcher.rb

```diff
@@ -116,12 +138,19 @@ def with_message(message)
           self
         end
 
+        def allow_nil(allow_nil = true)
+          @allow_nil = allow_nil
+          self
+        end
+
         def matches?(subject)
           super(subject)
           @expected_message ||= :blank
 
           if secure_password_being_validated?
             disallows_and_double_checks_value_of!(blank_value, @expected_message)
+          elsif @allow_nil
+            allows_value_of(nil)
```

Comment:
@mcmire struggling a bit here to elegantly test both association AND simple attribute because `disallows_value_of(blank_value, @expected_message)` fails due to a [ActiveRecord association type mismatch](https://github.com/rails/rails/blob/v4.0.0/activerecord/lib/active_record/associations/association.rb#L213) for empty string.

Also realizing that I'm not even validating that the `belongs_to` record exists.  This seems like a big effort.

Plus I'm also getting concerned that this is in the `active_model` directory and not `active_record`.